spark/src/main/scala/org/apache/spark/sql/delta/commands/CreateDeltaTableCommand.scala

```diff
@@ -559,13 +564,34 @@ case class CreateDeltaTableCommand(
       table.storage.copy(properties = Map.empty)
     }
 
-    table.copy(
-      schema = new StructType(),
-      properties = Map.empty,
-      partitionColumnNames = Nil,
-      // Remove write specific options when updating the catalog
-      storage = storageProps,
-      tracksPartitionsInCatalog = true)
+    // If we have to update the catalog, use the correct schema and table properties, otherwise
+    // empty out the schema and property information
+    if (conf.getConf(DeltaSQLConf.DELTA_UPDATE_CATALOG_ENABLED)) {
+      // In the case we're creating a Delta table on an existing path and adopting the schema
+      val schema = if (table.schema.isEmpty) snapshot.schema else table.schema
+      val truncatedSchema = UpdateCatalog.truncateSchemaIfNecessary(schema)
+      val additionalProperties = if (truncatedSchema.isEmpty) {
+        Map(UpdateCatalog.ERROR_KEY -> UpdateCatalog.LONG_SCHEMA_ERROR)
+      } else {
+        Map.empty
+      }
+
+      table.copy(
+        schema = truncatedSchema,
+        partitionColumnNames = Nil,
```

Comment:
Could we add code comments around why we don't pass `partitionColumnNames` even when the delta table is partitioned?